src/drafts/MarkdownEditor/_useListEditing.ts

```diff
@@ -60,9 +60,9 @@ export const parseListItem = (line: string): ListItem | null => {
 }
 
 export const listItemToString = (item: ListItem) =>
-  `${item.leadingWhitespace}${typeof item.delimeter === 'number' ? `${item.delimeter}.` : item.delimeter}${
-    item.middleWhitespace
-  }${item.taskBox || ''} ${item.text}`
+  typeof item.delimeter === 'number'
+    ? `${item.leadingWhitespace}${`${item.delimeter}.`}${item.middleWhitespace}${item.text}`
```

Comment:
[nit] extra string nesting is unnecessary here
```suggestion
    ? `${item.leadingWhitespace}${item.delimeter}.${item.middleWhitespace}${item.text}`
```